internal/core/metadatahandle.cpp

```diff
@@ -65,9 +85,17 @@ SingleMetadataHandle::SingleMetadataHandle(OpenVDS::VolumeDataLayout const* cons
     {
     validate_dimensionality(layout->GetDimensionality());
 
-    validate_minimal_nsamples(this->m_iline);
-    validate_minimal_nsamples(this->m_xline);
-    validate_minimal_nsamples(this->m_sample);
+    std::unordered_map<AxisType, Axis> axes_map;
+
+    for (int dimension = 0; dimension < layout->GetDimensionality(); ++dimension) {
+        auto name = std::string(layout->GetDimensionName(dimension));
```

Comment:
I was actually not thinking about using C-strings but rather if we could accomplish this somehow by using enums or similar provided by OpenVDS rather than comparing strings.

Reply:
The problem is that openvds provides us this information as C-strings (layout->DimensionName). They don't have enums for such things, only we do.

So I think the only way to get this information into some kind of enum is to compare C-strings or std::strings(C-string).

Reply:
That answers my question then 😸